include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -151,6 +151,54 @@ struct walk_adjacent_difference
     }
 };
 
+// the C++ stuff types to distinct "init vs. no init"
+template <typename _InitType>
+struct __init_value
+{
+    _InitType __value;
+    using __value_type = _InitType;
+};
+
+template <typename _InitType = void>
+struct __no_init_value
+{
+    using __value_type = _InitType;
+};
+
+// structure for the correct processing of the initial scan element
+template <typename _InitType>
+struct __init_processing
+{
+    template <typename _Tp>
+    void
+    operator()(const __init_value<_InitType>& __init, _Tp&& __value) const
+    {
+        __value = __init.__value;
```

Comment:
Should we forward `__value`? Question is relevant to all similar places in that file

Reply:
I don't change the code approach. I would to create separate issue to check the all similar places in that file.